drivers/adodb-pdo_firebird.inc.php

```diff
@@ -0,0 +1,423 @@
+				$this->SetFetchMode($savem);
+			}
+			$ADODB_FETCH_MODE = $save;
+			return $false;
+		}
+
+		$indexes = array();
+		while ($row = $rs->FetchRow()) {
+			$index = $row[0];
+			if (!isset($indexes[$index])) {
+				if (is_null($row[3])) {
+					$row[3] = 0;
+				}
+				$indexes[$index] = array(
+					'unique' => ($row[3] == 1),
+					'columns' => array()
+				);
+			}
+			$sql = "SELECT * FROM RDB\$INDEX_SEGMENTS WHERE RDB\$INDEX_NAME = '".$index."' ORDER BY RDB\$FIELD_POSITION ASC";
+			$rs1 = $this->Execute($sql);
+			while ($row1 = $rs1->FetchRow()) {
+				$indexes[$index]['columns'][$row1[2]] = $row1[1];
+			}
+		}
+		// restore fetchmode
+		if (isset($savem)) {
+			$this->SetFetchMode($savem);
+		}
+		$ADODB_FETCH_MODE = $save;
+
+		return $indexes;
+	}
+
+	public function metaPrimaryKeys($table,$owner_notused=false,$internalKey=false)
+	{
+		if ($internalKey) {
+			return array('RDB$DB_KEY');
+		}
+
+		$table = strtoupper($table);
+
+		$sql = 'SELECT S.RDB$FIELD_NAME AFIELDNAME
+	FROM RDB$INDICES I JOIN RDB$INDEX_SEGMENTS S ON I.RDB$INDEX_NAME=S.RDB$INDEX_NAME
+	WHERE I.RDB$RELATION_NAME=\''.$table.'\' and I.RDB$INDEX_NAME like \'RDB$PRIMARY%\'
+	ORDER BY I.RDB$INDEX_NAME,S.RDB$FIELD_POSITION';
+
+		$a = $this->GetCol($sql,false,true);
+		if ($a && sizeof($a)>0) return $a;
+		return false;
+	}
+
+	public function createSequence($seqname = 'adodbseq', $startID = 1)
+	{
+		$ok = $this->execute("CREATE SEQUENCE $seqname");
+		if (!$ok)
+			return false;
+
+		return $this->execute("ALTER SEQUENCE $seqname RESTART WITH ".($startID-1));
+	}
+
+	public function dropSequence($seqname = 'adodbseq')
+	{
+		$seqname = strtoupper($seqname);
+		return $this->Execute("DROP SEQUENCE $seqname");
+	}
+
+
+	public function _affectedrows()
+	{
+		return fbird_affected_rows( $this->_transactionID ? $this->_transactionID : $this->_connectionID );
+	}
+
+	public function genId($seqname='adodbseq',$startID=1)
+	{
+		$getnext = ("SELECT Gen_ID($seqname,1) FROM RDB\$DATABASE");
+		$rs = @$this->execute($getnext);
+		if (!$rs) 
+		{
+			$this->execute(("CREATE SEQUENCE $seqname" ));
+			$this->execute("ALTER SEQUENCE $seqname RESTART WITH ".($startID-1).';');
+			$rs = $this->execute($getnext);
+		}
+		if ($rs && !$rs->EOF) {
+			$this->genID = (integer) reset($rs->fields);
+		}
+		else {
+			$this->genID = 0; // false
+		}
+
+		if ($rs) {
+			$rs->Close();
+		}
+
+		return $this->genID;
+	}
+
+	public function selectLimit($sql,$nrows=-1,$offset=-1,$inputarr=false, $secs=0)
+	{
+		$nrows = (integer) $nrows;
+		$offset = (integer) $offset;
+		$str = 'SELECT ';
+		if ($nrows >= 0) $str .= "FIRST $nrows ";
+		$str .=($offset>=0) ? "SKIP $offset " : '';
+
+		$sql = preg_replace('/^[ \t]*select/i',$str,$sql);
+		if ($secs)
+			$rs = $this->cacheExecute($secs,$sql,$inputarr);
+		else
+			$rs = $this->execute($sql,$inputarr);
+
+		return $rs;
+	}
+
+	/**
+	* Sets the appropriate type into the $fld variable
+	*
+	* @param	string		$fld		By reference
```

Comment:
Looking at the function call at line 95, it would seem that $fld is an `ADOFieldObject`, not a string